### PR TITLE
coll/libnbc: add bruck algorithm for MPI_Iallgather

### PR DESCRIPTION
**Pull request by alex-anenkov**
Implements Bruck algorithm for MPI_Iallgather.

Signed-off-by: Alex Anenkov <anenkov.ru@gmail.com>

### CODE REVIEW

**Comment by bosilca** [comment]
ok to test.

**Comment by alex-anenkov** [comment]
bot:ompi:retest

File: ompi/mca/coll/libnbc/nbc_iallgather.c [hunk 1]
```diff
@@ -87,18 +96,44 @@ static int nbc_allgather_init(const void* sendbuf, int sendcount, MPI_Datatype s
     return res;
   }
 
+  /* allocate temporary buffers */
```
[the remark this answers is not in the document]
Comment (alex-anenkov) [reply ompi/mca/coll/libnbc/nbc_iallgather.c]: I agree, that this approach is ugly, but I adhered overall design of libnbc (allreduce, alltoall, etc). I am ready to fix this PR, but it is required an additional discussion of memory allocation in libnbc.

**Comment by awlauria** [comment]
Ping - now almost a year old. Any update on this pr?

**Comment by awlauria** [comment]
Adding this to the 5.0 milestone for tracking. Since this seems more like a feature than a bug fix, if you want it for 5.0 please target getting this into master by the end of April. Thanks.

**Comment by gpaulsen** [comment]
@alex-anenkov - How was this tested?  Is this safe for v5.0?

**Comment by alex-anenkov** [comment]
Hello @gpaulsen. Two years have passed since the opening of this PR. I'm not involved with OpenMPI development, so I'm not sure if PR is safe for v5.0. I think it will be better if I close it. Thanks.

**Comment by gpaulsen** [comment]
@alex-anenkov That's a shame, but thank you.

**Pull request closed without merging**